src/Qir/Runtime/lib/QIR/QirOutputHandling.cpp

```diff
@@ -15,9 +16,61 @@ static void PrintCStr(const char* cStr)
     WriteToCurrentStream(&msg);
 }
 
+static std::string EscapeString(std::string const& s)
+{
+    std::string out = "";
+    for (std::string::const_iterator i = s.begin(), end = s.end(); i != end; i++)
+    {
+        char c = *i;
```

Comment:
Consider using the [range-based `for` loop](https://github.com/AnthonyCalandra/modern-cpp-features#range-based-for-loops), it is considered less error-prone than the classical `for` loop.
```suggestion
    for (auto c: s)
    {
```

Reply:
done